src/plugins/transform/sources.rs

```diff
@@ -57,15 +59,15 @@ impl PluginTransform for Plugin {
         // Update each service to point to our locally cloned sources.
         let project = ctx.project;
         for service in &mut file.services.values_mut() {
-            for sources_result in service.sources(project.sources())? {
-                let (mount_as, source) = sources_result?;
+            for source_mount in service.sources(project.sources())? {
+                let source = source_mount.source;
                 if source.is_available_locally(project) && source.mounted() {
```

Comment:
This is the second time I've seen `source.is_available_locally(project) && source.mounted()` in pre-existing `cage` code near your diffs. I'm wondering why it doesn't have a single function combining the two, even if it has a boring name like `source.is_available_locally_and_mounted()`. We need to distinguish between "theoretically mounted according to our stored flags, except the source is missing" and "actually mounted", and I don't know what to name them.

Obviously, this isn't _your_ problem. :-) But I wouldn't mind hearing your input on this.

Reply:
Yeah, looking across the whole codebase, I think there are 4 different places that do this. But then there are also places that check one of them individually, or check for one of them true and the other false. I wonder if an enum could help here, to provide a more semantic wrapper around those two different flags/states, which could then potentially be made internal state that others don't need to care about?

Reply:
Yeah, you're exactly right that this should be an enum. Enums are so simple, but they're actually probably one of my favorite parts of Rust because  they bring so much clarity to cases like this.  Also, code that calls just one of the two needs a careful audit to see why it isn't calling the other.

Thank you for your input!